sdk/core/System.ClientModel/api/System.ClientModel.net6.0.cs

```diff
@@ -67,6 +67,19 @@ public partial class ClientPipelineOptions
         public System.ClientModel.Primitives.PipelineTransport? Transport { get { throw null; } set { } }
         public void AddPolicy(System.ClientModel.Primitives.PipelinePolicy policy, System.ClientModel.Primitives.PipelinePosition position) { }
     }
+    public partial class HttpClientPipelineTransport : System.ClientModel.Primitives.PipelineTransport, System.IDisposable
+    {
+        public static readonly System.ClientModel.Primitives.HttpClientPipelineTransport Shared;
+        public HttpClientPipelineTransport() { }
+        public HttpClientPipelineTransport(System.Net.Http.HttpClient client) { }
+        protected override System.ClientModel.Primitives.PipelineMessage CreateMessageCore() { throw null; }
+        public void Dispose() { }
+        protected virtual void Dispose(bool disposing) { }
+        protected virtual void OnReceivedResponse(System.ClientModel.Primitives.PipelineMessage message, System.Net.Http.HttpResponseMessage httpResponse) { }
```

Comment:
Are we exposing the HttpResponseMessage/HttpRequestMessage types because they are needed by Azure.Core in order to derive from this type?

Reply:
That is correct.  You can see where we're using these extensibility points in Azure.Core to add Azure-specific features here: https://github.com/Azure/azure-sdk-for-net/blob/feature/core-experiment/sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.Adapter.cs#L26-L53

Let me know if you have any concerns about this, or if this was just a clarifying question.